kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/impl/cache/Reflector.java

```diff
@@ -220,6 +220,9 @@ class ReflectorWatcher implements Watcher<T> {
 
     @Override
     public void eventReceived(Action action, T resource) {
+      if (stopFuture.isDone()) {
+        return;
+      }
```

Comment:
I'm fine with this change, but not sure if it might be breaking/unexpected for some users.

Just worried that some users might report that events are missing.

_I don't think K8s emits resource-related events to confirm/ack the Watch has been closed, so shouldn't be a problem_

Reply:
I don't think that would be an issue.  Those are events that are processed after stop has been called.